### PR TITLE
doc: tutorial refactor

### PR DESCRIPTION
**Pull request by mshabunin**
* compact lists of tutorials - moved information about original author and compatible OpenCV version into tutorial, removed brief description
* combined videoio, highgui and imgcodecs into new `app` section (application utils), flattened content
* combined photo, stitching, video, objedetect and ml into new `other` section, flattened content
* flattened structure of `imgproc` tutorials
* kept old pages for links compatibility
* created script `doc/tools/scan_tutorials.py` which checks order in tutorial lists and prev/next navigation links for consistency (must be run in the `doc` folder)
* :new: added tableofcontents to most tutorials

[preview](https://pullrequest.opencv.org/buildbot/export/pr/18712/docs/d9/df8/tutorial_root.html)

Note: have not updated G-API tutorials yet to avoid possible conflicts

Note: USAC tutorial is not added to any list and have slightly different format than others, I think it would be better to fix it in separate PR

```
force_builders_only=docs
```

### CODE REVIEW

**Comment by mshabunin** [comment]
`core` section before:
image
...and after:
image

**Comment by mshabunin** [comment]
Typical tutorial header...
image
I tried to combine prev/next links with author/version information, but didn't find nice variant.

**Comment by catree** [comment]
Is it possible to have also previous/next tutorial links at the bottom of the page?

In my opinion, it is useful when you follow a tutorial to not have to scroll up to go to the next tutorial.

Some examples:
- [BehaviorTree.CPP tutorials](https://www.behaviortree.dev/tutorial_02_basic_ports/)
- [ROS tutorials](http://wiki.ros.org/ROS/Tutorials/NavigatingTheFilesystem) (more subtle, the link is in the last sentence)
- [MoveIt tutorials](http://docs.ros.org/en/melodic/api/moveit_tutorials/html/doc/motion_planning_api/motion_planning_api_tutorial.html)

With the MoveIt tutorials, I like the previous/next buttons at the bottom of the page and the "address bar" at the top of the page, that could be used in OpenCV to navigate at the different levels of the tutorials hierarchy.

One issue with previous/next links is that it is not automatic. So if you want to add a tutorial, you have to update these links manually for two consecutive tutorials.

**Comment by mshabunin** [comment]
@catree , thank you for comments. I've added TOCs to most tutorials.

As for prev/next links at the bottom and better navigation, I don't see a convenient way to do it without manually copying this part from the top. We need to either extend tutorial-checking script added in this PR to be able to modify all tutorials, either some JS code which would transform contents at loading time. I'll try to experiment with the first approach later.

Languages information and tutorial description was removed in this PR, because in most cases this information is redundant and not very useful. I think maybe we can use longer titles if current one is not descriptive enough.

**Comment by alalek** [comment]
> `@tableofcontents`

Looks like it doesn't work on public CI. Which doxygen version is required?

**Comment by sturkmen72** [comment]
i thing updating doxigen version will be fine.

Last Release is [1.8.20](https://www.doxygen.nl/manual/changelog.html)

**Comment by catree** [comment]
@alalek 

It should work. There are already some pages that have table of contents:

- https://pullrequest.opencv.org/buildbot/export/pr/18712/docs/db/dfa/tutorial_transition_guide.html
- https://docs.opencv.org/4.5.0/db/dfa/tutorial_transition_guide.html

In `tutorial_transition_guide.html`, `@tableofcontents` is after `@prev_tutorial` and `@next_tutorial`. So maybe that is the reason?

**Comment by mshabunin** [comment]
I use latest doxygen 1.8.20 and TOCs work for me. However I do not recommend upgrading yet because language-switching buttons do not work in many tutorials because H2 headers have changed to H1 and our JS code does not handle it well and must be updated.

I suggest leaving them in this PR, because they do not harm and can be useful in future when we upgrade or find another solution.

File: doc/tutorials/gapi/face_beautification/face_beautification.markdown [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Implementing a face beautification algorithm with G-API {#tutorial_gapi_face_beautification}
 
+@prev_tutorial{tutorial_gapi_anisotropic_segmentation}
+
 [TOC]
```
Comment (alalek) [inline doc/tutorials/gapi/face_beautification/face_beautification.markdown:5]: BTW, Also there is another way to specify table of contents. [Page](http://pullrequest.opencv.org/buildbot/export/pr/18712/docs/d4/d48/tutorial_gapi_face_beautification.html)

But I agree that we need to check/fix headers on problematic pages.

**Review by alalek — APPROVED** [review]
Looks good to me! Thank you :+1:

**Pull request merged**